kythe/cxx/indexer/cxx/frontend.cc

```diff
@@ -348,6 +348,7 @@ void IndexerContext::LoadDataFromUnpackedFile(
   file_data.mutable_info()->set_path(source_file_name);
   file_data.set_content(source_data.str());
   job.virtual_files.push_back(std::move(file_data));
+  job.unit.add_source_file(source_file_name);
```

Comment:
Ultimately, it's the job of the thing constructing the compilation unit to set the source file.  CUs coming from the extractor should have this field set appropriately, so unconditionally adding it here is most likely to result in two identical source files being present.

Reply:
That's true, but I think in this case the CLI indexer is simulating the extractor as well as indexing.